src/Uuid/Uuid.php

```diff
@@ -0,0 +1,334 @@
+	 * recursively if needed
+	 */
+	public function clear(bool $recursive = false): bool
+	{
+		// For all models with children: if $recursive,
+		// also clear UUIDs from cache for all children
+		if (
+			$recursive === true &&
+			$this->model &&
+			method_exists($this->model, 'children') === true
+		) {
+			foreach ($this->model->children() as $child) {
+				$child->uuid()->clear(true);
+			}
+		}
+
+		return Uuids::cache()->remove($this->key());
+	}
+
+	/**
+	 * Generator function for the local context
+	 * collection, which takes priority when looking
+	 * up the UUID/model from index
+	 * @internal
+	 */
+	final public function context(): Generator
+	{
+		yield from $this->context ?? [];
+	}
+
+	/**
+	 * Look up Uuid in cache and resolve
+	 * to identifiable model object.
+	 * Implemented on child classes.
+	 *
+	 * @codeCoverageIgnore
+	 */
+	protected function findByCache(): Identifiable|null
+	{
+		throw new LogicException('UUID class needs to implement the ::findByCache() method');
+	}
+
+	/**
+	 * Look up Uuid in local and global index
+	 * and return the identifiable model object.
+	 * Implemented on child classes.
+	 *
+	 * @codeCoverageIgnore
+	 */
+	protected function findByIndex(): Identifiable|null
+	{
+		throw new LogicException('UUID class needs to implement the ::findByIndex() method');
+	}
+
+	/**
+	 * Shorthand to create instance
+	 * by passing either UUID or model
+	 */
+	final public static function for(
+		string|Identifiable $seed,
+		Collection|null $context = null
+	): static {
+		if (is_string($seed) === true) {
+			return match (Str::before($seed, '://')) {
+				'page'   => new PageUuid(uuid: $seed, context: $context),
+				'file'   => new FileUuid(uuid: $seed, context: $context),
+				'site'   => new SiteUuid(uuid: $seed, context: $context),
+				'user'   => new UserUuid(uuid: $seed, context: $context),
+				// TODO: activate for uuid-block-structure-support
+				// 'block'  => new BlockUuid(uuid: $seed, context: $context),
+				// 'struct' => new StructureUuid(uuid: $seed, context: $context),
+				default  => throw new InvalidArgumentException('Invalid UUID URI: ' . $seed)
+			};
+		}
+
+		return match (true) {
+			$seed instanceof Page
+				=> new PageUuid(model: $seed, context: $context),
+			$seed instanceof File
+				=> new FileUuid(model: $seed, context: $context),
+			$seed instanceof Site
+				=> new SiteUuid(model: $seed, context: $context),
+			$seed instanceof User
+				=> new UserUuid(model: $seed, context: $context),
+			// TODO: activate for uuid-block-structure-support
+			// $seed instanceof Block
+			// 	=> new BlockUuid(model: $seed, context: $context),
+			// $seed instanceof StructureObject
+			// 	=> new StructureUuid(model: $seed, context: $context),
+			default
+			=> throw new InvalidArgumentException('UUID not supported for: ' . get_class($seed))
+		};
+	}
+
+	/**
+	 * Generate a new ID string
+	 */
+	final public static function generate(int $length = 15): string
```

Comment:
I suggest to use a default length of 16 because 128 bits are a common field length for ID fields (could be relevant for custom cache drivers).

```suggestion
	final public static function generate(int $length = 16): string
```